CONTRIBUTING.md

```diff
@@ -41,27 +47,30 @@ feature requests.
 
 If you would like to propose a new feature for the project then it would be
 best to first discuss your idea with the community to gauge their level of
-interest. You can use any of the communitication channels to have this
-discussion, but ideally a new github issue should be opened so that the
+interest. You can use any of the communication channels to have this
+discussion, but ideally a new GitHub issue should be opened so that the
 history of the discussions can be saved within the repository itself.
 The issue should include information about the requirements and
-usecases that it is trying to address.
+use cases that it is trying to address.
 
 If you would like to also work on the implementation of the feature then
 it should include a discussion of the proposed design and technical details
 of how it will be implemented.
 
+> Before checking in a new feature, you are expected to contribute full documentation for this
```

Comment:
Shriram wanted it in blinking red lights with an auto-bullhorn. So I went with the subtler indent. It's worth pointing out since writing docs is 'challenging' for many devs...